src/main/java/org/openrewrite/java/testing/assertj/SimplifyHasSizeAssertion.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.assertj;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.TypeMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+import java.util.Collections;
+
+public class SimplifyHasSizeAssertion extends Recipe {
+
+    private static final MethodMatcher ASSERT_THAT_MATCHER = new MethodMatcher("org.assertj.core.api.Assertions assertThat(..)");
+    private static final MethodMatcher HAS_SIZE_MATCHER = new MethodMatcher("org.assertj.core.api.* hasSize(int)");
+    private static final String HAS_SAME_SIZE_AS = "hasSameSizeAs";
+
+    private static final TypeMatcher CHAR_SEQUENCE_TYPE_MATCHER = new TypeMatcher("java.lang.CharSequence", true);
+    private static final TypeMatcher ITERABLE_TYPE_MATCHER = new TypeMatcher("java.lang.Iterable", true);
+    private static final TypeMatcher MAP_TYPE_MATCHER = new TypeMatcher("java.util.Map", true);
```

Comment:
Rather than matching the type of the argument, I think it would be better to match the methods we're after here:
`CharSequence#length()`
`Iterable#size()`
`Map#size()`

That ought to then fix the failing test I'd just added, and remove the need for the null check on the expression select.

Reply:
Good catch!
Thanks for the suggestions!